
message Frequency {
	required string value = 1;
}


message LastChange {
	required int32 timestamp = 2;
}


message ImageInfo {
	required string extension = 1;
	required int32 length = 2;
	required int32 chunk_count = 3;
}

message ImageChunk {
	required bytes data = 1;
}


message InProgress {
}


message Response {
	enum Type {
		FREQUENCY = 1;
		LAST_CHANGE = 2;
		IMAGE_INFO = 3;
		IMAGE_CHUNK = 4;
		IMAGE_CHANGING = 5;
		IMAGE_NONE = 6;
		BAD_REQUEST = 7;
	}
		
	required Type type = 1;

	oneof value {
		Frequency frequency = 2;
		LastChange last_change = 3;
		ImageInfo image_info = 4;
		ImageChunk image_chunk = 5;
		InProgress in_progress = 6;		
	}
}

